driver/src/orderbook/auction_data_reader.rs

```diff
@@ -183,12 +185,58 @@ impl PaginatedAuctionDataReader {
     }
 }
 
+pub struct IndexedAuctionDataReader(AuctionDataReader);
```

Comment:
I think this abstraction is baller! It all fit super well together in the end :rocket: